go/sqltypes/type.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sqltypes
+
+import (
+	"fmt"
+
+	"github.com/youtube/vitess/go/vt/proto/query"
+)
+
+// These bit flags can be used to query on the
+// common properties of types.
+const (
+	IsNumber   = int(query.Flag_ISNUMBER)
+	IsUnsigned = int(query.Flag_ISUNSIGNED)
+	IsFloat    = int(query.Flag_ISFLOAT)
+	IsQuoted   = int(query.Flag_ISQUOTED)
+	IsText     = int(query.Flag_ISTEXT)
+	IsBinary   = int(query.Flag_ISBINARY)
+)
+
+// Vitess data types. These are idiomatically
+// named synonyms for the query.Type values.
+const (
+	Null      = query.Type_NULL
+	Int8      = query.Type_INT8
+	Uint8     = query.Type_UINT8
+	Int16     = query.Type_INT16
+	Uint16    = query.Type_UINT16
+	Int24     = query.Type_INT24
+	Uint24    = query.Type_UINT24
+	Int32     = query.Type_INT32
+	Uint32    = query.Type_UINT32
+	Int64     = query.Type_INT64
+	Uint64    = query.Type_UINT64
+	Float     = query.Type_FLOAT
+	Double    = query.Type_DOUBLE
+	Timestamp = query.Type_TIMESTAMP
+	Date      = query.Type_DATE
+	Time      = query.Type_TIME
+	Datetime  = query.Type_DATETIME
+	Year      = query.Type_YEAR
+	Decimal   = query.Type_DECIMAL
+	Text      = query.Type_TEXT
+	Blob      = query.Type_BLOB
+	VarChar   = query.Type_VARCHAR
+	VarBinary = query.Type_VARBINARY
+	Char      = query.Type_CHAR
+	Binary    = query.Type_BINARY
+	Bit       = query.Type_BIT
+	Enum      = query.Type_ENUM
+	Set       = query.Type_SET
+)
+
+// bit-shift the mysql flags by one byte so we
+// can merge them with the mysql types.
+const (
+	mysqlUnsigned = 32 << 8
```

Comment:
Yeah. Playing with bits is always unreadable. These flags are actually used only against mysql types, and their max val is 255. So, we're safe.
I'm basically trying to build an "a.b" key using field & flags. This is to avoid a 2-level map. I think it's worth the efficiency because this is low level code.
I'll add better comments.